plugins/include/amxmisc.inc

```diff
@@ -897,3 +897,31 @@ stock get_playersnum_ex(GetPlayersFlags:flags = GetPlayers_None, const team[] =
 	get_players_ex(_, PlayersNum, flags, team);
 	return PlayersNum;
 }
+
+/**
+ * Precaches a player model file.
+ *
+ * @note Can only be used inside of the plugin_precache() forward.
+ * @note Also searches for a "T.mdl" file and precaches that one as well if found.
+ * @note Player models MUST be placed in the following directory:
+ *       "models/player/%name%/%name%.mdl" where "name" is the name of the model.
+ *
+ * @param name  Name of the model file without the .mdl extension (e.g. "admin")
+ * @param id    Variable to store the "T.mdl" cache id if found
+ *
+ * @return      Unique cache id of the model
+ * @error       If called outside of the plugin_precache() forward, an error is
+ *              thrown from the precache_model() native.
+ */
+stock precache_player_model(const name[], &id = 0)
+{
+	new model[128];
+	formatex(model, charsmax(model), "models/player/%s/%sT.mdl", name, name);
+
+	if(file_exists(model))
+		id = precache_generic(model);
+
+	static const extension[] = "T.mdl";
```

Comment:
why static ?

Reply:
@justgo97 It doesn't need to be defined more than once. It acts like a global variable with a local scope.

Reply:
static is bad here because its lifetime (or "extent") is the entire run of the program while these stock will be only use in one forward.

Reply:
`static` is fully OK for local array/string constants. There is no other option:
1. `new` isn't appropriate here because it does copying on each function call.
2. Global constant is bad because we use this constant only in this function, so there is no reason for it to be global.

Reply:
yeah but this string constant is pretty small in size and in its use having it static means it will stay in memory all the time while its only needed at the 1-2 seconds of map start for plugin_precache forward ?

Reply:
@justgo97 it will stay in memory with both `new` and `static`, but `new` will also copy data from "global" constant to local on each call.

Reply:
oh, i thought with new it will be deleted after local scope is finished.

Reply:
so new is better for memory in this case ?

Reply:
@justgo97 no, `new = data + stack`, `static = data`.

Reply:
@WPMGPRoSToTeMa I mean in overall which is better for memory ?

```
public plugin_init()
{
    func1()

    func2()
}

public func1()
{
    new const Block1[] = "I_M_GOING_TO_STAY_IN_MEMORY_JUST_FOR_THIS_FUNC"
    new const Block2[] = "I_M_GOING_TO_STAY_IN_MEMORY_JUST_FOR_THIS_FUNC"
    new const Block3[] = "I_M_GOING_TO_STAY_IN_MEMORY_JUST_FOR_THIS_FUNC"
    new const Block4[] = "I_M_GOING_TO_STAY_IN_MEMORY_JUST_FOR_THIS_FUNC"
    new const Block5[] = "I_M_GOING_TO_STAY_IN_MEMORY_JUST_FOR_THIS_FUNC"
    new const Block6[] = "I_M_GOING_TO_STAY_IN_MEMORY_JUST_FOR_THIS_FUNC"
}

public func2()
{
    static const Block1[] = "I_M_GOING_TO_STAY_IN_MEMORY_FOR_EVER"
    static const Block2[] = "I_M_GOING_TO_STAY_IN_MEMORY_FOR_EVER"
    static const Block3[] = "I_M_GOING_TO_STAY_IN_MEMORY_FOR_EVER"
    static const Block4[] = "I_M_GOING_TO_STAY_IN_MEMORY_FOR_EVER"
    static const Block5[] = "I_M_GOING_TO_STAY_IN_MEMORY_FOR_EVER"
    static const Block6[] = "I_M_GOING_TO_STAY_IN_MEMORY_FOR_EVER"
}
```

Reply:
@justgo97 if consider only permanent memory, then they are both the same.

Reply:
so for non-permanent memory new is better, which is an advantage for new over static ?

Reply:
It doesn't matter.